### ISSUE TITLE
Repl

### ISSUE THREAD

**Issue by heades** [heades/ILL-Impl#17]
Implement repl.  See grady implementation for inspiration.

**Comment by tattymennis** [comment]
Should I program a top-level let expression for the Repl? It would be very useful in testing (and using) the language, but might conflict with the let term parsers. If so, I could change the reserved "let" to be "letu" and "lett" respectively.

**Comment by tattymennis** [comment]
Gotcha, I know how I'll do it.

**Comment by tattymennis** [comment]
I've implemented the ability for the Repl to take in a top-level let variable and print it from the internal queue. However, I need to handle the error if that variable isn't in the queue. My intuition is to return with an Either String Term and case split on the output of the function, but I'm having conceptual difficulties of using Either within a REPLExpr or REPLStateIO(). I'm familiar with Either and simple monads, but I don't fully understand the intricacies of working in the environment.

Can we go over some more advanced monad topics on Tuesday?

**Comment by heades** [comment]
Yes, we can go over this on Tuesday.  Unfortunately, I do not have time to meet
until then.  Sorry.  Talk to you more on Tuesday.

**Comment by tattymennis** [comment]
No problem, sounds great. Thanks!
